src/commands/openCurrentBranchOnRemote.ts

```diff
@@ -0,0 +1,44 @@
+import { TextEditor, Uri, window } from 'vscode';
+import { Commands } from '../constants';
+import type { Container } from '../container';
+import { GitUri } from '../git/gitUri';
+import { RemoteResourceType } from '../git/remotes/provider';
+import { Logger } from '../logger';
+import { RepositoryPicker } from '../quickpicks/repositoryPicker';
+import { command, executeCommand } from '../system/command';
+import { ActiveEditorCommand, getCommandUri } from './base';
+import { OpenOnRemoteCommandArgs } from './openOnRemote';
+
+@command()
+export class OpenCurrentBranchOnRemoteCommand extends ActiveEditorCommand {
+	constructor(private readonly container: Container) {
+		super(Commands.OpenCurrentBranchOnRemote);
+	}
+
+	async execute(editor?: TextEditor, uri?: Uri) {
+		uri = getCommandUri(uri, editor);
+
+		const gitUri = uri != null ? await GitUri.fromUri(uri) : undefined;
+
+		const repository = await RepositoryPicker.getBestRepositoryOrShow(gitUri, editor, 'Open Current Branch Name');
```

Comment:
Should I be using `getBestRepositoryOrShow` or should I simply use `this.container.git.getBestRepository(editor)`?

Reply:
`getBestRepositoryOrShow`, so that if there isn't a clear repo, it will show a quick pick menu to the user to pick the appropriate repo

Reply:
thanks!